src/validation.cpp

```diff
@@ -3040,7 +3047,7 @@ bool ContextualCheckBlock(const CBlock& block, CValidationState& state, const Co
     //   {0xaa, 0x21, 0xa9, 0xed}, and the following 32 bytes are SHA256^2(witness root, witness nonce). In case there are
     //   multiple, the last one is used.
     bool fHaveWitness = false;
-    if (VersionBitsState(pindexPrev, consensusParams, Consensus::DEPLOYMENT_SEGWIT, versionbitscache) == THRESHOLD_ACTIVE) {
+    if (VersionBitsState(pindexPrev, consensusParams, Consensus::DEPLOYMENT_SEGWIT2X, versionbitscache) == THRESHOLD_ACTIVE) {
```

Comment:
@tomasvdw The text of the agreement does not require a HF to be activated at the same time as segwit. In fact, it doesn't require segwit to be activated at all. The signatories only committed to reducing the threshold for activation, although many - like BitFury - have indicated that they understand the agreement as segwit activates first, with a HF later.

Remember that the agreement also doesn't specify how the HF will be activated - being a HF a simple flag date would suffice, as it's the creation of a new currency.